main.go

```diff
@@ -222,3 +234,54 @@ func getCloudFoundryRedisBindings() (addrs, passwords, aliases []string) {
 
 	return
 }
+
+// loadRedisSentinel gets Redis IP addresses and Ports and return proper information.
+func loadRedisSentinel(addr, password, alias, separator string) ([]string, []string, []string, error) {
+	passwords = strings.Split(password, separator)
+	aliases = strings.Split(alias, separator)
+
+	options := []redis.DialOption{
+		redis.DialConnectTimeout(5 * time.Second),
+		redis.DialReadTimeout(5 * time.Second),
+		redis.DialWriteTimeout(5 * time.Second),
+	}
+
+	sentinel_addrs := strings.Split(addr, separator)
+	sntnl := &sentinel.Sentinel{Addrs: sentinel_addrs, MasterName: *masterName,
+		Dial: func(addr string) (redis.Conn, error) {
+			c, err := redis.Dial("tcp", addr, options...)
+			if err != nil {
+				return nil, err
+			}
+			return c, nil
+		},
+	}
+
+	var addrs []string
+	master, err := sntnl.MasterAddr()
+	if err != nil {
+		log.Warnln("Error while getting address of redis master", err)
+		return nil, nil, nil, err
+	} else {
+		addrs = append(addrs, master)
+
+		slaves, err := sntnl.Slaves()
+		if err != nil {
+			log.Warnln("Error while getting addresses of redis slaves", err)
+			return nil, nil, nil, err
+		} else {
+			for _, slave := range slaves {
+				if true == slave.Available() {
+					addrs = append(addrs, slave.Addr())
+					for len(passwords) < len(addrs) {
```

Comment:
This sets the password (and then in the next for loop) the aliases for all the discovered hosts to the same, right? 

Might move this down to right before line 284 to do it only once and also change it to `passwords = append(passwords, password)` (same for aliases) to make it more clear that they;re all set to the password/alias passed into this function.